data_safe_haven/infrastructure/programs/sre/workspaces.py

```diff
@@ -215,34 +186,13 @@ def __init__(
             "vm_outputs": vm_outputs,
         }
 
-    def read_cloudinit(
-        self,
-        apt_proxy_server_hostname: str,
-        ldap_group_filter: str,
-        ldap_group_search_base: str,
-        ldap_server_hostname: str,
-        ldap_server_port: str,
-        ldap_user_filter: str,
-        ldap_user_search_base: str,
-        software_repository_hostname: str,
-        storage_account_data_private_sensitive_name: str,
-        storage_account_data_private_user_name: str,
-    ) -> str:
+    @staticmethod
+    def template_cloudinit(**kwargs: str) -> str:
+        logger = get_logger()
         with open(
             resources_path / "workspace" / "workspace.cloud_init.mustache.yaml",
             encoding="utf-8",
         ) as f_cloudinit:
-            mustache_values = {
-                "apt_proxy_server_hostname": apt_proxy_server_hostname,
-                "ldap_group_filter": ldap_group_filter,
-                "ldap_group_search_base": ldap_group_search_base,
-                "ldap_server_hostname": ldap_server_hostname,
-                "ldap_server_port": ldap_server_port,
-                "ldap_user_filter": ldap_user_filter,
-                "ldap_user_search_base": ldap_user_search_base,
-                "software_repository_hostname": software_repository_hostname,
-                "storage_account_data_private_user_name": storage_account_data_private_user_name,
-                "storage_account_data_private_sensitive_name": storage_account_data_private_sensitive_name,
-            }
-            cloudinit = chevron.render(f_cloudinit, mustache_values)
-            return b64encode(cloudinit)
+            cloudinit = chevron.render(f_cloudinit, kwargs)
+            logger.debug(f"Generated cloud-init config:\n {cloudinit}")
```

Comment:
I'm not particularly worried. There aren't any secrets there (at the moment). It would also only expose these to a user who can read all of the secrets already.

Reply:
OK, and do we want a multiline log message here? I thought we were trying to avoid those.